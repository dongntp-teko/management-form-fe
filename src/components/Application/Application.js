// Created by thanhpd on 6/17/2019
// @flow
import React, { useEffect, useState } from 'reactn';

import { Table, Divider, Icon, Card, Button, Input } from 'antd';

import axios from 'axios';
import { applicationService } from 'services/application';
import Create from '../Create/Create';
import Update from '../Update/Update';
import Delete from '../Delete/Delete';

export default (props: Object) => {
  axios.defaults.baseURL = 'http://103.126.156.66:9000/api/';
  const [apps, setApps] = useState([{}]);
  const [visible, setVisible] = useState(false);
  const [loading, setLoading] = useState(false);
  const [stt, setStt] = useState({
    action: '',
    appId: '',
  });

  const getDataFromServer = value => {
    setLoading(true);
    // console.log('searchApplication')
    applicationService
      .searchApplication({
        app_name: value,
      })
      .then(response => {
        const { data } = response.data;
        setApps(data);
      })
      .catch(error => {
        console.log(error);
      })
      .then(() => {
        setLoading(false);
      });
  };

  const searchName = async value => {
    getDataFromServer(value);
  };

  useEffect(() => {
    getDataFromServer();
  }, []);

  const openModal = (value, id) => {
    setVisible(true);
    setStt({ action: value, appId: id });
  };

  const closeModal = () => {
    setVisible(false);
    setStt({ action: '', appId: '' });
    props.history.push('/app');
  };

  const columns = [
    {
      title: 'ID',
      dataIndex: 'app_id',
    },
    {
      title: 'Group',
      dataIndex: 'group_name',
    },
    {
      title: 'Name',
      dataIndex: 'app_name',
    },
    {
      title: 'Main Uri',
      dataIndex: 'main_uri',
    },
    {
      title: 'Timezone',
      dataIndex: 'timezone',
    },
    {
      title: 'Type',
      dataIndex: 'app_type',
    },
    {
      title: 'ecommerce',
      dataIndex: 'ecommerce',
      render: text =>
        !text ? (
          <Icon type="close" className="color-red" />
        ) : (
          <Icon type="check" className="color-green" />
        ),
    },
    {
      title: 'Site search',
      dataIndex: 'site_search',
      render: text =>
        !text ? (
          <Icon type="close" className="color-red" />
        ) : (
          <Icon type="check" className="color-green" />
        ),
    },
    {
      title: 'Keep Url Fragment',
      dataIndex: 'keep_url_fragment',
      render: text =>
        !text ? (
          <Icon type="close" className="color-red" />
        ) : (
          <Icon type="check" className="color-green" />
        ),
    },
    {
      title: '',
      render: (text, record) => {
        return (
          <div>
            <Button
              type="link"
              onClick={() => openModal('update', record.app_id)}
            >
              Update
            </Button>
            <Divider type="vertical" />
            <Button
              type="link"
              onClick={() => openModal('delete', record.app_id)}
            >
              Delete
            </Button>
          </div>
        );
      },
    },
  ];

  return (
    <div>
      <div>
        <h1>Dashboard</h1>
        <div className="mb10 flex-space-between">
          <Input.Search
            id="input-search"
            placeholder="Search name"
            onSearch={value => searchName(value)}
            style={{ width: 200 }}
          />
          <Button type="primary" onClick={() => openModal('create')}>
            <Icon type="plus" /> Create new app
          </Button>
        </div>

        {stt.action === 'create' && (
          <Create visible={visible} closeModal={closeModal} />
        )}
        {stt.action === 'update' && (
          <Update visible={visible} closeModal={closeModal} appId={stt.appId} />
        )}
        {stt.action === 'delete' && (
          <Delete visible={visible} closeModal={closeModal} appId={stt.appId} />
        )}
      </div>
      <Card>
        <Table
          id="table_app"
          rowKey="app_id"
          dataSource={apps.filter(app => app.activeness)}
          columns={columns}
          loading={loading}
          pagination={false}
        />
      </Card>
    </div>
  );
};
