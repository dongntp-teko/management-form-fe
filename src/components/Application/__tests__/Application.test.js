import React from 'react';
import { mount } from 'enzyme';
import TestRenderer from 'react-test-renderer';
import Application from '../Application';

describe('ThemeList', () => {
  it('should match snapshot', async done => {
    const container = TestRenderer.create(<Application />);
    expect(container.toJSON()).toMatchSnapshot();
    done();
  });

  describe('test', () => {
    test('data table should match reponse', async done => {
      console.log(123);
      const container = mount(<Application />);
      // await new Promise(resolve => setTimeout(resolve, 0));
      container.update();
      const table = container
        .find('table')
        .find('thead')
        .find('tr')
        .find('th');

      // const tr = table.findByType('tbody').findByType('tr')
      console.log(table);

      expect(table.length).toBe(10);

      console.log(table.map(item => item.text()));
      done();
    });
  });

  // describe('Functional', () => {
  //   describe('could handle form fields', () => {
  //     let findById;
  //     categoryServices.getMethods = jest.fn(() =>
  //       Promise.resolve({ data: [{ id: 4 }] }),
  //     );

  //     // beforeEach(async () => {
  //     //   axiosMocker.onGet({ data: { content: [{ id: 1 }], totalRecord: 1 } });

  //     //   const container = TestRenderer.create(
  //     //     <MemoryRouter>
  //     //       <ThemeList />
  //     //     </MemoryRouter>,
  //     //   );
  //     //   await axiosResponse();

  //     //   findById = id => container.root.findByProps({ id });
  //     // });

  //     it('onChange Agent', done => {
  //       const globalState = {
  //         paymentAgents: [{ id: 1 }, { id: 2 }],
  //         paymentChannels: [{ id: 2, agentId: 1 }],
  //         paymentServices: [{ id: 3 }],
  //       };
  //       setGlobal({ ...globalState });

  //       findById('channelId').props.onChange(2);
  //       expect(findById('agentId').props.value).toEqual(undefined);

  //       findById('agentId').props.onChange(1);
  //       expect(findById('channelId').props.value).toEqual(2);
  //       findById('agentId').props.onChange(2);
  //       expect(findById('channelId').props.value).toEqual(undefined);

  //       done();
  //     });

  //     it('could change active record', async done => {
  //       const updateTheme = jest.spyOn(themeServices, 'updateTheme');
  //       axiosMocker.onPut({ data: {} });

  //       const confirmActiveRecord = findById('confirm-active-record-1');
  //       confirmActiveRecord.props.onConfirm();
  //       await axiosResponse();

  //       expect(updateTheme).toBeCalledWith({
  //         benchId: 1,
  //         isActive: true,
  //       });
  //       done();
  //     });
  //   });

  //   describe('could handle table change', () => {
  //     let findByClass;
  //     categoryServices.getMethods = jest.fn(() =>
  //       Promise.resolve({ data: [{ id: 4 }] }),
  //     );

  //     beforeEach(async () => {
  //       axiosMocker.onGet({ data: { content: [], totalRecord: 12 } });

  //       const container = TestRenderer.create(
  //         <MemoryRouter>
  //           <ThemeList />
  //         </MemoryRouter>,
  //       );
  //       await axiosResponse();

  //       findByClass = className => container.root.findByProps({ className });
  //     });

  //     it('onChange Table', async done => {
  //       const searchTheme = jest.spyOn(themeServices, 'searchTheme');

  //       axiosMocker.onGet({ data: { content: [], totalRecord: 12 } });

  //       findByClass(' ant-pagination-next').props.onClick();
  //       await axiosResponse();

  //       expect(searchTheme).toBeCalledWith({ page: 1, size: 10 });
  //       done();
  //     });
  // });
  // });
});
