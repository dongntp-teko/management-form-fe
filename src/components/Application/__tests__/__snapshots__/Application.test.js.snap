// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeList should match snapshot 1`] = `
<div>
  <div>
    <h1>
      Dashboard
    </h1>
    <div
      className="mb10 flex-space-between"
    >
      <span
        className="ant-input-search ant-input-affix-wrapper"
        style={
          Object {
            "width": 200,
          }
        }
      >
        <input
          className="ant-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Search name"
          style={null}
          type="text"
          value=""
        />
        <span
          className="ant-input-suffix"
        >
          <i
            aria-label="icon: search"
            className="anticon anticon-search ant-input-search-icon"
            onClick={[Function]}
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </i>
        </span>
      </span>
      <button
        className="ant-btn ant-btn-primary"
        onClick={[Function]}
        type="button"
      >
        <i
          aria-label="icon: plus"
          className="anticon anticon-plus"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </i>
        <span>
           Create new app
        </span>
      </button>
    </div>
  </div>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-table-wrapper"
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
          >
            <div
              className="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left"
              id="table_app"
              style={Object {}}
            >
              <div
                className="ant-table-content"
              >
                <div
                  className="ant-table-body"
                  onScroll={[Function]}
                  onWheel={[Function]}
                  style={Object {}}
                >
                  <table
                    className=""
                    style={Object {}}
                  >
                    <colgroup>
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                    </colgroup>
                    <thead
                      className="ant-table-thead"
                    >
                      <tr
                        style={
                          Object {
                            "height": null,
                          }
                        }
                      >
                        <th
                          className=""
                        >
                          <span
                            className="ant-table-header-column"
                          >
                            <div>
                              <span
                                className="ant-table-column-title"
                              >
                                ID
                              </span>
                              <span
                                className="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span
                            className="ant-table-header-column"
                          >
                            <div>
                              <span
                                className="ant-table-column-title"
                              >
                                Group
                              </span>
                              <span
                                className="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span
                            className="ant-table-header-column"
                          >
                            <div>
                              <span
                                className="ant-table-column-title"
                              >
                                Name
                              </span>
                              <span
                                className="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span
                            className="ant-table-header-column"
                          >
                            <div>
                              <span
                                className="ant-table-column-title"
                              >
                                Main
                              </span>
                              <span
                                className="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span
                            className="ant-table-header-column"
                          >
                            <div>
                              <span
                                className="ant-table-column-title"
                              >
                                Timezone
                              </span>
                              <span
                                className="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span
                            className="ant-table-header-column"
                          >
                            <div>
                              <span
                                className="ant-table-column-title"
                              >
                                Type
                              </span>
                              <span
                                className="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span
                            className="ant-table-header-column"
                          >
                            <div>
                              <span
                                className="ant-table-column-title"
                              >
                                ecommerce
                              </span>
                              <span
                                className="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span
                            className="ant-table-header-column"
                          >
                            <div>
                              <span
                                className="ant-table-column-title"
                              >
                                Site search
                              </span>
                              <span
                                className="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span
                            className="ant-table-header-column"
                          >
                            <div>
                              <span
                                className="ant-table-column-title"
                              >
                                Keep Url Fragment
                              </span>
                              <span
                                className="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span
                            className="ant-table-header-column"
                          >
                            <div>
                              <span
                                className="ant-table-column-title"
                              >
                                
                              </span>
                              <span
                                className="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      className="ant-table-tbody"
                    />
                  </table>
                </div>
                <div
                  className="ant-table-placeholder"
                >
                  <div
                    className="ant-empty ant-empty-normal"
                  >
                    <div
                      className="ant-empty-image"
                    >
                      <img
                        alt="No Data"
                        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                      />
                    </div>
                    <p
                      className="ant-empty-description"
                    >
                      No Data
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
